### PR TITLE
refactor!: de-zarfify maru-runner

### LINKED ISSUE #23: Remove dependency on Zarf
### Description
Today Maru pulls in Zarf for various action cmds. Remove the dependency on Zarf and migrate actions commands to this repo

### LINKED ISSUE #60: De-dup types
### Describe what should be investigated or refactored

Now that https://github.com/defenseunicorns/zarf/issues/2395 has been completed, de-dup the types that we copied over

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This PR removes Zarf as a dependency of Maru, and proposes the following breaking changes to the library API surface:

1. Creating a simple registration system for `./` prefixed apps
2. Not reaching into os.Env within library code (i.e. when processing `RUN_<VAR>`)

This also proposes the following feature breaking changes:

1. Drop support for `files`.

## Related Issue

Fixes #23 
Fixes #60

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/maru-runner/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Comment by Racer159** [comment]
still wip, there is some breakage with files and this still depends on `message`

**Comment by UncleGedd** [comment]
Re: new deps. Looking at [defenseunicorns/pkg](https://github.com/defenseunicorns/pkg), and I see `helpers` but not `exec` or `variables` . Where are these deps coming from?

**Comment by UncleGedd** [comment]
I understand the registration system and removal of `config.CmdPrefix`, but I don't get the second breaking change: "Not reaching into os.Env within library code". Can we get more context there?

**Comment by Racer159** [comment]
> I understand the registration system and removal of `config.CmdPrefix`, but I don't get the second breaking change: "Not reaching into os.Env within library code". Can we get more context there?

This is a breakage for any users of Maru as a library - (which @ericwyles pointed out I was misinformed about `uds-cli`'s integration of Maru) - basically we only pull the `os.Env` in the `cmd` package now so if you use `Run` directly you'd have to inject those env vars as set variables yourself.

We should discuss but I also think that `uds-cli` should switch from vendoring Maru to calling `Run` so that it can have more control and we don't need to have global `config` vars for all of the configuration.

**Comment by UncleGedd** [comment]
> We should discuss but I also think that `uds-cli` should switch from vendoring Maru to calling `Run` so that it can have more control and we don't need to have global `config` vars for all of the configuration.

👍 I'm good with that

**Comment by Racer159** [comment]
> > We should discuss but I also think that `uds-cli` should switch from vendoring Maru to calling `Run` so that it can have more control and we don't need to have global `config` vars for all of the configuration.
> 
> 👍 I'm good with that

After discussing with @ericwyles yesterday I have changed my opinion back - we could do things to make it easier to run from library code (i.e. centralizing command flags) but they likely aren't necessary near term for what UDS CLI actually needs (likely a config var thing would work and we still should be able to control it well enough that vendoring apps like zarf / uds-cli don't step on each other in the globals)

**Review by ericwyles — CHANGES_REQUESTED** [review]

**Comment by Racer159** [comment]
Thinking of leaving prompt in for now (as dead code we can either cleanup or implement later) - I implemented generics though for the Zarf specific variable values that are carried through.

File: src/pkg/runner/actions.go [hunk 1]
```diff
@@ -261,21 +245,75 @@ func (r *Runner) performZarfAction(action *zarfTypes.ZarfComponentAction) error
...
+		cfg.Shell = *a.Shell
+	}
 
-	// Try to patch the binary path in case the name isn't exactly "./run".
-	prefix := "./run "
-	if config.CmdPrefix != "" {
-		prefix = fmt.Sprintf("./%s ", config.CmdPrefix)
+	// Add variables to the environment.
+	for k, v := range vars {
+		// Remove # from env variable name.
```
[the remark this answers is not in the document]
Comment (Racer159) [reply src/pkg/runner/actions.go]: Plan is to just have them set these as extra variables for now - we can work with Zarf to improve this in the future.

File: src/cmd/root.go [hunk 2]
```diff
@@ -92,16 +79,30 @@ func cliSetup() {
...
 		}
 	}
 
-	if !config.SkipLogFile && !ListTasks && !ListAllTasks {
+	if !skipLogFile && !listTasks && !listAllTasks {
 		utils.UseLogFile()
 	}
+
+	// Register `uds` and `zarf` as coming from the $PATH when used directly from the CLI
+	exec.RegisterCmdMutation("uds", "uds")
```
[the remark this answers is not in the document]
Comment (Racer159) [reply src/cmd/root.go]: done

**Review by ericwyles — APPROVED** [review]

**Pull request merged**